Detectors/AOD/include/AODProducerWorkflow/AODProducerWorkflowSpec.h

```diff
@@ -193,12 +197,20 @@ class AODProducerWorkflowDPL : public Task
   void findMinMaxBc(gsl::span<const o2::ft0::RecPoints>& ft0RecPoints, gsl::span<const o2::dataformats::PrimaryVertex>& primVertices, const std::vector<o2::InteractionTimeRecord>& mcRecords);
   uint64_t getTFNumber(uint64_t firstVtxGlBC, int runNumber);
 
-  template <typename TTracks, typename TTracksCursor, typename TTracksCovCursor, typename TTracksExtraCursor>
-  void fillTracksTable(const TTracks& tracks, std::vector<int>& vCollRefs, const TTracksCursor& tracksCursor,
-                       const TTracksCovCursor& tracksCovCursor, const TTracksExtraCursor& tracksExtraCursor, int trackType);
-
-  template <typename TTracks, typename TTracksCursor>
-  void fillMFTTracksTable(const TTracks& tracks, std::vector<int>& vCollRefs, const TTracksCursor& tracksCursor);
+  template <typename TMFTCursor, typename TTracksCursor, typename TTracksCovCursor, typename TTracksExtraCursor>
+  void fillTracksTable(gsl::span<const o2::its::TrackITS>& tracksITS,
+                       gsl::span<const o2::mft::TrackMFT>& tracksMFT,
+                       gsl::span<const o2::dataformats::TrackTPCITS>& tracksITSTPC,
+                       gsl::span<const o2::tpc::TrackTPC> tracksTPC,
```

Comment:
note that we will have ~15 different track types and they will be all available from the reco container as e.g. `recoContainer::getSpan<o2::its::TrackITS>(GlobalTrackID::ITS)` or via aliases `getITSTracks()` etc., so better to pass directly the RecoContainer.

Reply:
Hi @shahor02. Yes, I understand. I have made this PR only to implement track sorting and to test it separately. I can fix this after #6170 is merged.